src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -387,12 +390,22 @@ bool ShouldScanDependencies(Assembly assembly)
         return !IsExcluded(assemblyName.Name);
     }
 
+    // We are deliberately checking here against the MessageInterfaces assembly name because
+    // we the command, event and message interfaces have been moved there by using type forwarding.
+    // While it would be possible to read the type forwarding information from the assembly that imposes
+    // some performance overhead, and we don't expect that the assembly name will change nor that we will add many
+    // more type forwarding cases. Should that be the case we might want to revisit the idea of reading the metadata
+    // information from the assembly.
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    bool IsCoreOrMessageInterfaceAssembly(AssemblyName assemblyName) => assemblyName.Name == CoreAssemblyName || assemblyName.Name == MessageInterfacesAssemblyName;
```

Comment:
This should probably be an Ordinal comparison:
```suggestion
bool IsCoreOrMessageInterfaceAssembly(AssemblyName assemblyName) => string.Equals(assemblyName.Name, CoreAssemblyName, StringComparison.Ordinal) || string.Equals(assemblyName.Name, MessageInterfacesAssemblyName, StringComparison.Ordinal);
```

Reply:
I was hesitant to switch the comparison type, so I took the same comparison type as that was already in. string.Equals should probably be faster and safer here right?

Reply:
> string.Equals should probably be faster and safer here right?

I believe so, yes.